### PR TITLE
Issue #14911: Added a new section 4.8.5.3 in google style guide implementation

### PR DESCRIPTION
**Pull request by Zopsss**
issue #14911 

added a new section `4.8.5.3 Methods And Constructors Annotations` in google style guide coverage page and wrote test cases for it

### CODE REVIEW

**Comment by Zopsss** [comment]
GitHub, generate site

File: src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/InputMethodsAndConstructorsAnnotations.java [hunk 1]
```diff
@@ -0,0 +1,75 @@
+package com.google.checkstyle.test.chapter4formatting.rule4853methodsandconstructorsannotations;
+
+public class InputMethodsAndConstructorsAnnotations {
```
Comment (rdiachenko) [inline src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/InputMethodsAndConstructorsAnnotations.java:3]: > Exception: A single parameterless annotation may instead appear together with the first line of the signature

Two cases are missed:
1. Multiple annotations on the same line with the method signature. We expect a warning here.
2. A **single parameterised** annotation on the same line with the method signature. We expect a warning here.

Comment (Zopsss) [reply src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/InputMethodsAndConstructorsAnnotations.java:3]: Done. 

https://github.com/checkstyle/checkstyle/blob/859ba58c92f71d16f922479a2b228222b2cd5b0b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/InputMethodsAndConstructorsAnnotations.java#L41-L43

https://github.com/checkstyle/checkstyle/blob/859ba58c92f71d16f922479a2b228222b2cd5b0b/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/InputMethodsAndConstructorsAnnotations.java#L78-L80

I have considered both, method and constructor for this as it was not specifically mentioned which signature type they are talking about.

**Review by rdiachenko — APPROVED** [review]
lgtm

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/MethodsAndConstructorsAnnotationsTest.java [hunk 2]
```diff
@@ -0,0 +1,76 @@
...
+import com.google.checkstyle.test.base.AbstractGoogleModuleTestSupport;
+import com.puppycrawl.tools.checkstyle.api.Configuration;
+import com.puppycrawl.tools.checkstyle.checks.annotation.AnnotationLocationCheck;
+import com.puppycrawl.tools.checkstyle.checks.javadoc.InvalidJavadocPositionCheck;
+
+public class MethodsAndConstructorsAnnotationsTest extends AbstractGoogleModuleTestSupport {
+
+    @Override
+    protected String getPackageLocation() {
+        return "com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations";
```
Comment (romani) [inline src/it/java/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/MethodsAndConstructorsAnnotationsTest.java:36]: please wrap long String value to multiline.
and remove  all suppressions.

File: config/checkstyle-non-main-files-suppressions.xml [hunk 3]
```diff
@@ -60,6 +60,8 @@
   <!-- test resources are weird by design, no validations in them -->
   <suppress id="lineLength"
          files="src[\\/]it[\\/]resources[\\/].*[\\/]InputLineLength.*\.java"/>
+  <suppress id="lineLength"
+            files="src[\\/]it[\\/]resources[\\/].*[\\/]InputMethodsAndConstructorsAnnotations.*\.java"/>
```
Comment (romani) [inline config/checkstyle-non-main-files-suppressions.xml:64]: this is not ok,
please wrap all code in Inputs to not be lengthy.

File: src/xdocs/google_style.xml [hunk 4]
```diff
@@ -1408,6 +1408,43 @@
...
+                  </span>
+                  <a href="checks/annotation/annotationlocation.html#AnnotationLocation">
+                    AnnotationLocation</a>
+                  <br />
+                  <br />
+                  <span class="wrapper inline">
+                    <img src="images/ok_green.png" alt="" />
+                  </span>
+                  <a href="checks/javadoc/invalidjavadocposition.html#InvalidJavadocPosition">
+                    InvalidJavadocPosition</a>
```
Comment (romani) [inline src/xdocs/google_style.xml:1435]: why javadoc Check appears here ?
Chapter does not care on javadoc.

**Review by romani — CHANGES_REQUESTED** [review]
items:

File: src/xdocs/google_style.xml (hunk 4 above)
Comment (Zopsss) [reply src/xdocs/google_style.xml:1435]: it does, here's the its usage:

https://github.com/checkstyle/checkstyle/blob/f82714823ef21f43fc3543de5706690ead752f0f/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/InputMethodsAndConstructorsAnnotations.java#L28-L33

https://github.com/checkstyle/checkstyle/blob/f82714823ef21f43fc3543de5706690ead752f0f/src/it/java/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/MethodsAndConstructorsAnnotationsTest.java#L53-L59

File: src/it/java/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/MethodsAndConstructorsAnnotationsTest.java (hunk 2 above)
Comment (Zopsss) [reply src/it/java/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/MethodsAndConstructorsAnnotationsTest.java]: done 

https://github.com/checkstyle/checkstyle/blob/b0dd2d65f8f163adef2fb6f9d2e6cc21152390c5/src/it/java/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/MethodsAndConstructorsAnnotationsTest.java#L35-L38

File: config/checkstyle-non-main-files-suppressions.xml (hunk 3 above)
Comment (Zopsss) [reply config/checkstyle-non-main-files-suppressions.xml]: done. removed all these suppressions. Wrapped code:

https://github.com/checkstyle/checkstyle/blob/b0dd2d65f8f163adef2fb6f9d2e6cc21152390c5/src/it/resources/com/google/checkstyle/test/chapter4formatting/rule4853methodsandconstructorsannotations/InputMethodsAndConstructorsAnnotations.java#L78-L79

File: src/xdocs/google_style.xml (hunk 4 above)
Comment (romani) [reply src/xdocs/google_style.xml:1435]: There is another chapter in style guide that demands proper locations for all javadocs. But wording of this chapter is not mentioned about javadoc anything.
Im this table we map wording to Checks, to let anyone of us in future look at this table and see how each phrase is related to used Checks.

Comment (romani) [reply src/xdocs/google_style.xml:1435]: Please share words from chapter that you think vis about location of javadoc

Comment (Zopsss) [reply src/xdocs/google_style.xml:1435]: From Filed Annotations' documentation:

> The rules for annotations on method and constructor declarations are the same as the [previous section](https://google.github.io/styleguide/javaguide.html#s4.8.5.2-class-annotation-style). Example:

here the previous section is Class Annotations. 

From Class Annotations' documentation:

> Annotations applying to a class appear immediately after the documentation block, and each annotation is listed on a line of its own (that is, one annotation per line).

This does talks about the documentation ( i.e javadoc ). We've used the InvalidJavadocPosition in that section so we're using at here as this section has same rules as it.

Comment (romani) [reply src/xdocs/google_style.xml:1435]: I see it now, thanks a lot !

**Review by romani — APPROVED** [review]
ok to merge

**Pull request merged**